### DISCUSSION TITLE
Better PDF handling.

### DISCUSSION THREAD

**Discussion by shetozz** [flxzt/rnote#519] [Ideas]
I'm using the flatpak version, so I'm not on the last commit or anything. 

But right now to properly use Rnote to annotate PDF I have to set the canvas to the appropriate size of the PDF -good luck if you have several page sizes-.

There are extra pages added to the original document, no support for invert color mode, and sometimes the PDF opens with wrong offset to the canvas.

There are some issues opened for some of these problems and I don't wanna make a duplicate of them, so I opened this discussion to avoid that.

Is there a possibility to fix some of these issue of it's out of scope?

**Comment by flxzt** [comment]
Regarding

> There are extra pages added to the original document

and

> sometimes the PDF opens with wrong offset to the canvas.

I would appreciate if you could file bug reports for that, I don't think these exist already.

There is already discussion about a better Pdf annotation mode in #153, it would be better to leave suggestions there. It is not out of scope, but as the nature of open-source hobby projects is, features and improvements might come slower depending on how much time and motivation I and other contributors have for it.

**Comment by shetozz** [comment]
> I would appreciate if you could file bug reports for that, I don't think these exist already.

This issue is the same as #429. 
It's not annoying if you have an infinite or simi-infinte canvas, but when you set the canvas to single page sometimes half of the page gets placed out of canvas, and you have to move the document manually.


![Screenshot from 2023-02-20 13-57-37](https://user-images.githubusercontent.com/93838524/220101695-f5bb687f-6070-4371-abb3-1b5d8ef5eaaf.png)


![Screenshot from 2023-02-20 13-58-05](https://user-images.githubusercontent.com/93838524/220101711-408a9594-ffa8-437b-9ed5-c7543f8272a8.png)

**Reply by flxzt** [reply]
ah ok, i see. You wrote **sometimes** so I assumed this is buggy behaviour.

**Reply by shetozz** [reply]
After writing the topic, I found out how to make this behavior disappear, if the normal PDF page size is A4 set the canvas to A3 and leave the scale of the PDF to a 100%, it will be kinda normal.

![Screenshot from 2023-02-20 14-30-57](https://user-images.githubusercontent.com/93838524/220108779-23a833a2-3c8c-43bb-8182-7c68195e2201.png)

**Reply by flxzt** [reply]
The offset is actually always relative to the current viewport, so that the pdf doesn't get inserted somewhere out of view. But I just ( bf4476e42e2820b514167897716dccf846bd5ac8 ) improved this offset so that the imported pdf doesn't land somewhere out of the page bounds anymore.

**Reply by shetozz** [reply]
Thanks for the fix.
